wallet/wallet-cli/src/main.rs

```diff
@@ -107,22 +109,37 @@ async fn run(output: &ConsoleContext) -> Result<(), WalletCliError> {
             .map_err(WalletCliError::WalletError)?
     };
 
-    let controller = wallet_controller::make_rpc_controller(
+    let mut controller = wallet_controller::make_rpc_controller(
+        chain_config,
         rpc_address,
         Some((&rpc_username, &rpc_password)),
         wallet,
     )
     .await
     .map_err(WalletCliError::Controller)?;
 
-    repl::start_cli_repl(output, controller, &data_dir, vi_mode).await
+    let (event_tx, event_rx) = mpsc::unbounded_channel();
+
+    let output_copy = output.clone();
+    let repl_command = repl::get_repl_command();
+    let create_line_editor =
+        repl::create_line_editor(repl_command.clone(), output, &data_dir, vi_mode)?;
+
+    std::thread::spawn(move || {
+        // Run a blocking loop in a separate thread
+        repl::run(create_line_editor, repl_command, event_tx, &output_copy);
+    });
```

Comment:
You should join this at the end of the program. We don't need more issues like p2p.